taichi/transforms/auto_diff.cpp

```diff
@@ -28,9 +200,9 @@ class ConvertLocalVar : public BasicStmtVisitor {
                          })
                          .empty();
     if (!load_only) {
-      alloc->replace_with(
-          Stmt::make<StackAllocaStmt>(alloc->ret_type.data_type, 16));
-      // TODO: remove 16 here
+      alloc->replace_with(Stmt::make<StackAllocaStmt>(
+          alloc->ret_type.data_type,
+          get_current_program().config.ad_stack_size));
```

Comment:
We can analyze the exact needed size in `cfg_optimization` in the future.

Reply:
Right, allocating stacks with compile-time known size will be very helpful for performance.